### PR TITLE
fix: fix usage of `#[abi(tag)]` attribute with elaborator

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

## Problem\*

Resolves 

## Summary\*

Followup to #5292 to handle globals as well

## Documentation\*

Check one:
- [ ] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by TomAFrench** [comment]
Also needs a struct example as well

File: test_programs/compile_success_contract/abi_attribute/src/main.nr [hunk 1]
```diff
@@ -0,0 +1,4 @@
+contract Foo {
+    #[abi(foo)]
+    global foo: Field = 42;
```
Comment (TomAFrench) [inline test_programs/compile_success_contract/abi_attribute/src/main.nr:3]:
```suggestion
    global foo: Field = 42;
    
    #[abi(bar)]
    struct Bar {
        inner: Field
    }
```

File: compiler/noirc_frontend/src/elaborator/mod.rs [hunk 2]
```diff
@@ -1190,7 +1190,7 @@ impl<'context> Elaborator<'context> {
         self.current_item = Some(DependencyId::Global(global_id));
         let let_stmt = global.stmt_def;
 
-        if !self.in_contract
+        if !self.module_id().module(self.def_maps).is_contract
```
Comment (jfecher) [inline compiler/noirc_frontend/src/elaborator/mod.rs:1193]: Maybe we should remove `self.in_contract` entirely if it is redundant?
I suppose there's the impl case where we override the value.

**Review by jfecher — APPROVED** [review]
Using `#[abi(foo)]` on a struct will also issue a warning currently I believe since it will attempt to look up a comptime function by that name to run on the struct. I wonder if we should differentiate "call this function" annotations from "mark this function".

**Comment by TomAFrench** [comment]
We need to check that these actually end up in the build artifact as I ran into situations where it seems like they were still being omitted in the sync PR. I've had to turn off the elaborator for now in aztec-packages as a result.

File: compiler/noirc_frontend/src/elaborator/mod.rs (hunk 2 above)
Comment (TomAFrench) [reply compiler/noirc_frontend/src/elaborator/mod.rs:1193]: Potentially, we can do this in a separate PR however.

**Comment by TomAFrench** [comment]
> I wonder if we should differentiate "call this function" annotations from "mark this function". It is either that or give up on the warning entirely which can be confusing if the annotation is misspelled I think.

I agree that we shouldn't just assume that any custom attribute is a function which needs to be called. Is that how we're handling it atm?

**Comment by jfecher** [comment]
> I agree that we shouldn't just assume that any custom attribute is a function which needs to be called. Is that how we're handling it atm?

If the function isn't callable we issue a warning that it wasn't found. This is only for "Custom" attributes. Otherwise it is very hard to tell the difference between an unknown word that is meant to be a marker and an unknown word that was a typo meant to call a function.

**Pull request merged**